File: lib/src/utils/logger.dart
import 'dart:io';

import 'package:flutter/foundation.dart';
import 'package:nearby_service/nearby_service.dart';

class Logger {
  Logger._();

  static NearbyServiceLogLevel level =
      kDebugMode ? NearbyServiceLogLevel.debug : NearbyServiceLogLevel.error;

  static void debug(String message) {
    if (level.index <= NearbyServiceLogLevel.debug.index) {
      debugPrint(
        _messageWrapper(
          message,
          androidColor: AndroidConsoleColor.grey,
          iosIcon: IOSConsoleIcon.settings,
        ),
      );
    }
  }

  static void info(String message) {
    if (level.index <= NearbyServiceLogLevel.info.index) {
      debugPrint(
        _messageWrapper(
          message,
          androidColor: AndroidConsoleColor.green,
          iosIcon: IOSConsoleIcon.success,
        ),
      );
    }
  }

  static void error(Object? error) {
    if (level.index <= NearbyServiceLogLevel.error.index) {
      debugPrint(
        _messageWrapper(
          '$error',
          androidColor: AndroidConsoleColor.red,
          iosIcon: IOSConsoleIcon.error,
        ),
      );
    }
  }

  static String _messageWrapper(
    String message, {
    required AndroidConsoleColor androidColor,
    required IOSConsoleIcon iosIcon,
  }) {
    if (Platform.isAndroid) {
      return '\x1B[${androidColor.value}m[NearbyService]: $message\x1B[0m';
    }
    if (Platform.isIOS) {
      return '[NearbyService ${iosIcon.value}]: $message ';
    }
    return message;
  }
}

enum AndroidConsoleColor {
  grey(37),
  green(32),
  red(31);

  const AndroidConsoleColor(this.value);

  final int value;
}

enum IOSConsoleIcon {
  settings('🛠'),
  success('✅'),
  error('❌');

  const IOSConsoleIcon(this.value);

  final String value;
}

File: android/src/main/kotlin/com/xenikii/nearby_service/Logger.kt
package com.xenikii.nearby_service

import android.util.Log

const val TAG = "NearbyService"

/**
 * Class for systemizing log levels.
 */
enum class LogLevel(val value: Int) {
    DEBUG(1),
    INFO(2),
    ERROR(3),
    DISABLED(4),
}

class Logger {
    companion object {
        var level = LogLevel.DEBUG
        fun d(message: String) {
            if (level.value <= LogLevel.DEBUG.value) {
                Log.d(TAG, "\u001B[37m$message\u001B[0m")
            }
        }

        fun i(message: String) {
            if (level.value <= LogLevel.INFO.value) {
                Log.i(TAG, "\u001B[32m$message\u001B[0m")
            }
        }

        fun e(message: String) {
            if (level.value <= LogLevel.ERROR.value) {
                Log.e(TAG, "\u001B[31m$message\u001B[0m")
            }
        }
    }
}